reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,173 @@
+
+To give to integrator a clean way to rely on your code, you can alse create interfaces.
+
+
+Extensibility
+-------------
+
+Respect Akeneo PIM dependencies
+"""""""""""""""""""""""""""""""
+You **MUST** follow Akeneo PIM dependencies at tall cost. Don't try to change version of existing ones.
+You may add new dependencies needed for your customisations.
+
+
+Use registries
+""""""""""""""
+Registries are also a good extension point when you have to manage with many cases (different attribute types, entities, etc.).
+It consists in creating a registry class and then tag services which will be add to this registry.
+
+.. note::
+    Learn more about `registry`_ pattern.
+
+You can also take example on the localizers' cookbook: :doc:`/cookbook/localization/how_to_use_the_localizers`
+
+.. _registry: http://martinfowler.com/eaaCatalog/registry.html
+
+
+Plug on events (datagrid, savers, remover, etc.)
+""""""""""""""""""""""""""""""""""""""""""""""""
+It is the best extensibility point you have.
+Unfortunately, it is not always possible as Akeneo PIM can't know where you want to plug your code.
+If you can not, it does not matter, you can decorate or extend your class and add the extension point you need.
+By the way, feel free to contribute opening an issue to discuss with our team on our `github repository`_
+or let us know your need on our `forum`_. It is important for us to improve ourselves and our application.
+
+.. _github repository: https://github.com/akeneo/pim-community-dev
+.. _forum: https://www.akeneo.com/fr/forums/
+
+
+Avoid to override Akeneo PIM classes/services
+"""""""""""""""""""""""""""""""""""""""""""""
+The easiest way to add a new behavior on your services is to override it.. But wait! That's not a good thing!
+First, each override leads to potential compatibility problems.
+Let's imagine that you override the `pim_catalog.saver.product` because you want to add a specific action when you save a product.
+Now, a final customer wants to use your bundle and another one which override the same service (Tough luck!).
+The last bundle defined in the `app/AppKernel.php` will defined the service used..
+And in both cases, one of the actions will not be applied.
+
+The second effect of overriding a parameter or a service is that it will have a side effect on the whole application.
+If you need to impact transversely Akeneo PIM and you can't plug on events, that's a way to do but keep in mind
+you can use create your own service with your own class.
+
+
+Models and Repositories
+"""""""""""""""""""""""
+You should avoid to extends the models (entities and documents) and repositories.
+The problem is the same than when you override classes/services.
+
+There is unfortunately no perfect solution for these cases.
+
+Don't extend an entity to add it properties.
+
+Two possibilities here:
+
+- There is not a lot of changes on the entity you overrode. Document what is missing and let's the integrators do it.
```

Comment:
I don't understand, aren't these best practices supposed to be actually read by integrators ? 
And it is "Let integrators"